### PR TITLE
Typehint and validate tags and status params to create_run

### PR DESCRIPTION
**Pull request by schrockn**
### Summary & Motivation

Typehint and check the tags and status parameters to create_run.

Tags in callsites comes from sources that are both Mapping[str, str] and Mapping[str, object]. Doing a validation pass to ensure that they are in fact str to str.

Status was relying on the fact that all callsites to create_run_for_pipeline were not respecting the typehint on its status parameter and were passing the enum instead.

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11446** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11446" target="_blank">Graphite</a> 
    * **PR #11447** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11447" target="_blank">Graphite</a> 
      * **PR #11448** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11448" target="_blank">Graphite</a> 
        * **PR #11449** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11449" target="_blank">Graphite</a> 
          * **PR #11450** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11450" target="_blank">Graphite</a> 
            * **PR #11451** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11451" target="_blank">Graphite</a> 
              * **PR #11452** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11452" target="_blank">Graphite</a> 
                * **PR #11453** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11453" target="_blank">Graphite</a> 
                  * **PR #11454** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11454" target="_blank">Graphite</a> 
                    * **PR #11455** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11455" target="_blank">Graphite</a> 
                      * **PR #11456** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11456" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11456?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/instance/__init__.py [hunk 1]
```diff
@@ -1234,7 +1240,7 @@ def create_run(
             solids_to_execute=solids_to_execute,
             step_keys_to_execute=step_keys_to_execute,
             status=status,
-            tags=tags,
+            tags=dict(validated_tags),
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/instance/__init__.py:1243]: should `_construct_run_with_snapshots` take `frozentags` instead? guess this `dict` call is harmless

**Review by alangenfeld — APPROVED** [review]
> Status was relying on the fact that all callsites to create_run_for_pipeline were not passing in any value for status, which is the only reason why that codepath was working.

should this PR drop `status` as an arg then? Not sure how that lines up with further refactoring, so just take it under consideration.

**Comment by schrockn** [comment]
Actually upon further inspection the reason why create_run_for_pipeline is working is that all of the callsites actually don't respect the typehint

File: python_modules/dagster/dagster/_core/instance/__init__.py (hunk 1 above)
Comment (schrockn) [reply python_modules/dagster/dagster/_core/instance/__init__.py:1243]: I don't want to muck with the code that mangles the tags in _construct_run_with_snapshots in this PR.

**Pull request merged**